core/src/Streamly/Internal/Unicode/Stream.hs

```diff
@@ -28,7 +28,12 @@ module Streamly.Internal.Unicode.Stream
     , decodeUtf8
     , decodeUtf8'
     , decodeUtf8_
+
+    -- ** UTF-16 Decoding
+    , decodeUtf16'
+    , decodeUtf16
```

Comment:
We can keep these unexposed. Don't know if we'll ever use this without taking byte-order into consideration.